### ISSUE TITLE
roxygen 7.0.0 causes oce doc warnings and errors

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1624]
An `update.packages()` on my R-devel machine bumped up the Roxygen version from 6.1.1 to 7.0.0, and now I get a build-check warnings and errors.

I'll look into this sometime today.  There may be a broken GH commit in the middle of the work, because that's how I'll transmit code to my other machine, which has the older Roxygen (and, possibly relevant, older R).

**Warnings**
```R
An object of \code{\link[base]{class}} \code{"oce"} that has been
```
and now we get
```R
An object of \verb{\link[base]\{class}} \code{"oce"} that has been
```
which gives the (oddly repetitive) warnings
```R
Warning: /Users/kelley/git/oce/man/decimate.Rd:30: unexpected macro '\code'
Warning: /Users/kelley/git/oce/man/decimate.Rd:30: All text must be in a section
Warning: /Users/kelley/git/oce/man/decimate.Rd:30: All text must be in a section
Warning: /Users/kelley/git/oce/man/decimate.Rd:31: All text must be in a section
```
and similar problems for `download.met()`.

**Errors**

```R
Warning: newline within quoted string at geodXy.Rd:93
Error in parse_Rd("/Users/kelley/git/oce/man/geodXy.Rd", encoding = c(Encoding = "UTF-8"),  : 
  Unexpected end of input (in " quoted string opened at geodXy.Rd:93:26)
```

**Comment by dankelley** [comment]
Re the "error", I think this is a bug in roxygen. I fixed the "error" by splitting a line and putting a space after a comma in argument list.  I think they must have a bad regexp or something in roxygen. Still, I don't see any problem with this slight rewriting, so I'll keep it.

With the error now gone, we get *a ton* of warnings.  I'm going to look into royxgen to see if this is a known bug or feature.

**Comment by dankelley** [comment]
Hm, I'm now thinking that these problems stem from a mixture of old and new notations in the Roxygen markup within oce, e.g. using an older (markup-laden) way to specify links, instead of the much cleaner `[function()]` way.   I'll go through things one by one.

It's even pointing out a problem with the very base docs at the start of `R/oce.R`, no less:
```R
Warning: [/Users/kelley/git/oce/R/oce.R:3] @description mismatched braces or quotes
Warning: [/Users/kelley/git/oce/R/oce.R:9] @details mismatched braces or quotes
```
I think this is because that doc didn't have more than one paragraph before new-name subsections, so it wasn't sure where to put those subsections.  I'll fix that by moving some text around etc.

**Comment by dankelley** [comment]
This is all cleaned up now. Yesterday, I got things working without error or warning on my R-devel machine with the new Roxygen2.  That build also produced no problems at test time, nor in travisCI automated testing.  Then, today, I did the same procedure on another machine with R-stable with the older Roxygen2 (again, all clear) and then I updated that machine to have the same R-stable but the new Roxygen2 (and, again, all clear).

What I've found is that the new roxygen2 catches more problems than the old one did, and that's great. But there are also some wierd things, e.g. it changes some backtick things into `\code{}` blocks and others into `\verb{}` blocks, and I'm not sure how it decides on that.  Another thing it does is to format `@family` things differently (one per line, compared to the old in paragraph style), and that can yield `git diff` entries that do not signal significant change.
